src/components/FaceDetector/FaceDetector.js

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect, useCallback } from 'react';
+import { useAppMessage } from '@daily-co/daily-react';
+import './FaceDetector.css';
+import FaceDetection from '../../utils/facedetection';
+
+export default function FaceDetector({ id, isLocal, videoRef }) {
+  const [detectionEnabled, setDetectionEnabled] = useState(false);
+  // eslint-disable-next-line no-unused-vars
+  const [detector, setDetector] = useState();
+  const [detectorInterval, setDetectorInterval] = useState();
+  const [result, setResult] = useState('');
+  const [remoteTracking, setRemoteTracking] = useState();
+
+  const sendAppMessage = useAppMessage({
+    onAppMessage: useCallback((ev) => {
+      if (ev.fromId === id && ev.data.face) {
```

Comment:
I think this may be more intuitive without repeating the fromId check.

Can we  just early out if `fromId  !== id` since nothing else seems to happen without it? Then we can just check for existence of `ev.data.face` and the relevant property if needed.